libpod/container_validate.go

```diff
@@ -128,8 +128,10 @@ func (c *Container) validate() error {
 
 	// If User in the OCI spec is set, require that c.config.User is set for
 	// security reasons (a lot of our code relies on c.config.User).
-	if c.config.User == "" && (c.config.Spec.Process.User.UID != 0 || c.config.Spec.Process.User.GID != 0) {
-		return errors.Wrapf(define.ErrInvalidArg, "please set User explicitly via WithUser() instead of in OCI spec directly")
+	if c.config.Spec != nil {
+		if c.config.User == "" && (c.config.Spec.Process.User.UID != 0 || c.config.Spec.Process.User.GID != 0) {
+			return errors.Wrapf(define.ErrInvalidArg, "please set User explicitly via WithUser() instead of in OCI spec directly")
```

Comment:
Is an end user going to know what WithUser() is?

Reply:
Not sure, this line was in there beforehand (I think?) I added the `c.config.Spec != nil` line so a de-reference error doesn't happen

Reply:
If this line pops up, whoever is writing code against Libpod messed up, not the user.